compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/UKernelOps.td

```diff
@@ -65,15 +65,33 @@ def IREECodegen_UKernelGenericOp :
     Variadic<AnyRankedTensorOrMemRefType>:$outputs,
     Variadic<AnyType>:$other_operands,
     OptionalAttr<DictionaryAttr>:$fn_def_attrs,
-    OptionalAttr<IndexAttr>:$strided_outer_dims);
+    OptionalAttr<IndexListArrayAttr>:$strided_dims);
   let results = (outs Variadic<AnyType>:$results);
   let assemblyFormat = [{
     attr-dict $u_kernel_fn_name
     (`ins` `(` $inputs^ `:` type($inputs) `)`)?
     (`outs` `(` $outputs^  `:` type($outputs) `)`)?
     (`(` $other_operands^ `:` type($other_operands) `)`)?
     ( `fn_def_attrs` $fn_def_attrs^ )?
-    (`strided_outer_dims` `(` $strided_outer_dims^ `)`)? (`->` type($results)^)?
+    (`strided_dims` `(` $strided_dims^ `)`)? (`->` type($results)^)?
+  }];
+
+  let builders = [
+    OpBuilder<(ins "TypeRange":$result_types, "StringRef":$u_kernel_fn_name,
+      "ValueRange":$inputs, "ValueRange":$outputs, "ValueRange":$other_operands,
+      "DictionaryAttr":$fn_def_attrs, "int64_t":$num_strided_outer_dims)>,
+    OpBuilder<(ins "TypeRange":$result_types, "StringRef":$u_kernel_fn_name,
+      "ValueRange":$inputs, "ValueRange":$outputs, "ValueRange":$other_operands,
+      "DictionaryAttr":$fn_def_attrs,
+      "ArrayRef<SmallVector<int64_t>>":$strided_dims)>
+  ];
+  let extraClassDeclaration = [{
+    // Get the number of strided outer dims for the given operand index.
+    // the strided_dims array contains the dims for only the ShapedType
+    // operands, so this returns the corresponding strided_dims value for
+    // the operand, or std::nullopt if the operand is not a ShapedType or
+    // strided_dims is null.
+    SmallVector<int64_t> getOperandStridedDims(int64_t operandIdx);
```

Comment:
I think the first sentence is more like `Returns the corresponding strided_dims value for the operand`?

btw, it is not `std::optional<>`. The comment seems outdated because you say that `std::nullopt` can be returned.